### PR TITLE
[2082] Fixed setting of rodsHostAddr_t for detached mode (main)

### PR DESCRIPTION
**Pull request by JustinKyleJames**
This fixes detached mode.  It should be released ASAP because detached mode has some significant issues.  (See #2082.).

### CODE REVIEW

File: s3/libirods_s3.cpp [hunk 1]
```diff
@@ -2142,46 +2143,54 @@ irods:: error s3StartOperation(irods::plugin_property_map& _prop_map)
...
+        if (resolveHost(&addr, &local_host) < 0) {
+            logger::resource::warn(
+                "[resource_name={}] Detached mode failed to set RESOURCE_HOST to {}."
+                "  Failed to call resolveHost.",
+                resource_name,
+                local_hostname);
+            return SUCCESS();
         }
+
+        irods::set_resource_property<rodsServerHost_t*>(resource_name, irods::RESOURCE_HOST, local_host);
```
Comment (korydraughn) [inline s3/libirods_s3.cpp:2190]: Does `local_host` result in a memory leak on destruction?

**Comment by JustinKyleJames** [comment]
Ping to take a look and merge.

File: s3/libirods_s3.cpp [hunk 2]
```diff
@@ -2142,46 +2143,54 @@ irods:: error s3StartOperation(irods::plugin_property_map& _prop_map)
...
+        rodsHostAddr_t addr{};
+        std::strncpy(addr.hostAddr, local_hostname, LONG_NAME_LEN);
+        const auto& zone_name = irods::get_server_property<const std::string>(irods::KW_CFG_ZONE_NAME);
+        std::strncpy(addr.zoneName, zone_name.c_str(), NAME_LEN);
+
+        rodsServerHost_t* local_host = nullptr;
+        if (resolveHost(&addr, &local_host) < 0) {
+            logger::resource::warn(
+                "[resource_name={}] Detached mode failed to set RESOURCE_HOST to {}."
+                "  Failed to call resolveHost.",
```
Comment (alanking) [inline s3/libirods_s3.cpp:2184]: Worth printing the error code here?

File: s3/libirods_s3.cpp [hunk 3]
```diff
@@ -2142,46 +2143,54 @@ irods:: error s3StartOperation(irods::plugin_property_map& _prop_map)
...
+                "  Failed to get irods::RESOURCE_NAME property.",
+                local_hostname);
+            return SUCCESS();
         }
 
-        if (error) {
-            // log the error but continue
-            s3_logger::error("[resource_name={}] Attached mode failed to set RESOURCE_HOST to {}.", resource_name.c_str(), resource_location);
+        // Look up the rodsHostAddr_t for this server and set this as the RESOURCE_HOST of the resource.
+        // Issue 2082: The rodsHostAddr_t entry is shared throughout the code (not just used for this
```
Comment (alanking) [inline s3/libirods_s3.cpp:2173]: Maybe don't refer to the issue number here?

File: s3/libirods_s3.cpp [hunk 4]
```diff
@@ -2142,46 +2143,54 @@ irods:: error s3StartOperation(irods::plugin_property_map& _prop_map)
...
+                "Detached mode failed to set RESOURCE_HOST to {}."
+                "  Failed to get irods::RESOURCE_NAME property.",
+                local_hostname);
+            return SUCCESS();
         }
 
-        if (error) {
-            // log the error but continue
-            s3_logger::error("[resource_name={}] Attached mode failed to set RESOURCE_HOST to {}.", resource_name.c_str(), resource_location);
+        // Look up the rodsHostAddr_t for this server and set this as the RESOURCE_HOST of the resource.
```
Comment (alanking) [inline s3/libirods_s3.cpp:2172]: Should this say "Look up the rodsServerHost_t..."?

**Review by alanking — COMMENTED** [review]
Sorry this fell off my radar. Overall the changes seem fine, just a few things that seem off to me.

I think more clarification in the commit message could potentially be helpful as well, but it would mainly be rehashing comments in the code and/or the linked issue. Up to you.

**Review by alanking — APPROVED** [review]

**Comment by trel** [comment]
seems good - squash em!

**Comment by JustinKyleJames** [comment]
> seems good - squash em!

done

**Pull request merged**